netlink-sys/src/tokio.rs

```diff
@@ -136,19 +100,20 @@ impl TokioSocket {
         cx: &mut Context,
         buf: &mut [u8],
     ) -> Poll<io::Result<(usize, SocketAddr)>> {
-        trace!("poll_recv_from called");
-        ready!(self.0.poll_read_ready(cx, mio::Ready::readable()))?;
-
-        trace!("poll_recv_from socket is ready for reading");
-        match self.0.get_ref().recv_from(buf, 0) {
-            Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
-                trace!("poll_recv_from socket would block");
-                self.0.clear_read_ready(cx, mio::Ready::readable())?;
-                Poll::Pending
-            }
-            x => {
-                trace!("poll_recv_from {:?} bytes read", x);
-                Poll::Ready(x)
+        loop {
```

Comment:
Why do we loop here? Shoudn't we simply return `Poll::Pending` on `WouldBlock`? I'm not sure.

Reply:
In tokio 0.2 we only need call `clear_read_ready()` and return `Poll::Pending`, the `clear_read_ready()` will register the current task to be notified once a read readiness event is received, but there is a big change in tokio 1.0, the corresponding method `clear_ready` called by `try_io` will not register the current task we need poll the AsyncFd again through the loop.

Reply:
Thanks for the explanation!

One more question: would simply clearing the readiness flag by calling [`clear_ready()`](https://docs.rs/tokio/1.0.2/tokio/io/unix/struct.AsyncFdReadyGuard.html#method.clear_ready) and returning `Poll::Pending` be enough to be notified of readiness, instead of looping?

Reply:
No, we must call `poll_read_ready()` after the `clear_ready() ` called.

The `clear_read_ready` in `PollEvented` will call the `poll_read_ready`, but the `clear_ready` in `AsyncFd` won't do that for us, we nee do it by ourselves.